### PR TITLE
CB-1447 S3Guard configuration managed at environment level

### PR DESCRIPTION
**Pull request by topolyai5**

### CODE REVIEW

File: environment/src/main/java/com/sequenceiq/environment/environment/domain/Environment.java [hunk 1]
```diff
@@ -347,4 +351,12 @@ public Tunnel getTunnel() {
     public void setTunnel(Tunnel tunnel) {
         this.tunnel = tunnel;
     }
+
+    public Json getParameters() {
```
Comment (attilapalfi92) [inline environment/src/main/java/com/sequenceiq/environment/environment/domain/Environment.java]: I don't like that we start adding jsons to the relational database when it's unavoidable. Instead add a pojo.

Comment (attilapalfi92) [reply environment/src/main/java/com/sequenceiq/environment/environment/domain/Environment.java]: Suggestion 2: if you can't let jsons go for some reason, than at least implement it like we did with the `EnvironmentTelemetry` here. Only this class has to know that it has a json representation inside the DB, in the code we want to do json conversions.

File: environment/src/main/java/com/sequenceiq/environment/environment/dto/EnvironmentDtoConverter.java [hunk 2]
```diff
@@ -45,13 +51,27 @@ public EnvironmentDto environmentToDto(Environment environment) {
...
+            builder.withAws(getParameterAs(environment, AwsEnvironmentParamsDto.class));
+        }
         if (environment.getNetwork() != null) {
             builder.withNetwork(environmentNetworkConverterMap.get(CloudPlatform.valueOf(environment.getCloudPlatform()))
                     .convertToDto(environment.getNetwork()));
         }
         return builder.build();
     }
 
+    private <T> T getParameterAs(Environment environment, Class<T> clss) {
```
Comment (attilapalfi92) [inline environment/src/main/java/com/sequenceiq/environment/environment/dto/EnvironmentDtoConverter.java]: It's not a very good solution to handle jsons, do it like we did with `EnvironmentTelemetry`.

File: environment/src/main/java/com/sequenceiq/environment/environment/domain/Environment.java (hunk 1 above)
Comment (attilapalfi92) [reply environment/src/main/java/com/sequenceiq/environment/environment/domain/Environment.java]: Suggestion 1 was this: https://stackoverflow.com/a/10434692/3757113

This is better than the json.

File: environment/src/main/java/com/sequenceiq/environment/environment/dto/EnvironmentDtoConverter.java (hunk 2 above)
Comment (attilapalfi92) [reply environment/src/main/java/com/sequenceiq/environment/environment/dto/EnvironmentDtoConverter.java]: Or what's even better: don't use json at all, for this single field. It's overkill and unnecessary. Use the relational database like it's designed to be used.

File: environment/src/main/java/com/sequenceiq/environment/environment/domain/Environment.java (hunk 1 above)
Comment (bergerdenes) [reply environment/src/main/java/com/sequenceiq/environment/environment/domain/Environment.java]: reworked

File: environment/src/main/java/com/sequenceiq/environment/environment/dto/EnvironmentDtoConverter.java (hunk 2 above)
Comment (bergerdenes) [reply environment/src/main/java/com/sequenceiq/environment/environment/dto/EnvironmentDtoConverter.java]: reworked

**Pull request merged**